Expand macros in version

benningm opened the pull request:
The version for the rpm is extracted from the spec file, but it
may contain macros which are not expanded by git-build-rpm.

This commit tries to get version from the rpmspec command which is
available in newer RPM versions:

  rpmspec -q --qf '%{version}' <specfile>

It will fall back to the old behaviour in case it fails, because
rpmspec is only shipped with rpm >= 4.9.

Additional it checks for unexpanded macros in the version string and
aborts build if macros are found.

Commit:
Expand macros in version

The version for the rpm is extracted from the spec file, but it
may contain macros which are not expanded by git-build-rpm.

This commit tries to get version from the rpmspec command which is
available in newer RPM versions:

  rpmspec -q --qf '%{version}' <specfile>

It will fall back to the old behaviour in case it fails, because
rpmspec is only shipped with rpm >= 4.9.

Additional it checks for unexpanded macros in the version string and
aborts build if macros are found.

theory commented:
Thank you, good idea. Question: Why the change to the regex matching the version line in the SPEC file at line 76?

benningm commented:
In my use case i want to get the version from the repository using my git-tag-version[1] command:

```
Version: %(git tag-version -l)
```

[1] https://metacpan.org/pod/distribution/Git-TagVersion/bin/git-tag-version

Commit:
Expand macros when rpmspec is not available

Expand the macros with rpm -E '<string>' in case rpm is unavailable.
(rpm before 4.9)

benningm commented:
I just extended the merge request to also expand macros in case rpmspec is not available using "rpm -E"

theory commented:
I tweaked your changes and pulled them in as fa0343d and 49e6531.
